### ISSUE TITLE
Warn before committing to protected branches

### ISSUE THREAD

**Issue by rhalaly** [microsoft/vscode#141867]
It can be helpful to warn before committing to commonly protected branches like `main`, `master`, `dev`, `release`, etc.
The warning can contain these actions "Create a new branch", "Commit anyway", and "Don't show me again for this repository".

**Comment by jgunnink** [comment]
Related to [62665](https://github.com/Microsoft/vscode/issues/62665)

I've been burned by this before, and have just been burned by it again now, so I went looking to see if this has been reported before and found this issue.

In today's case it was extremely fortunate that we had good knowledge of the changes, but this has potential to be extremely annoying/catastrophic. Looking at the history it seems that it ignores the protected branch that's on @github and uses some kind of forceful pushing mechanism to write to the main/master/trunk branch. Wondering if this could be changed as a short term fix to not be forceful and then a more eloquent solution, like the suggestion in the issue description or creating an array of protected branches in the settings or something.

**Comment by rhalaly** [comment]
Adding an array of branches names that should be protected to the workspace settings is a great idea.

Still, I would like to see a message that will allow me to create new branch instantly, skip the warning temporary, or disable the protection for this workspace or globally.

I agree that it should be balanced. It can be very annoying in some cases, like personal small projects that you want to commit the main branch directly, but to be protected when the repository defines protected branches.

**Comment by lszomoru** [comment]
Closing as duplicate of #148783.

**Comment by rhalaly** [comment]
Thanks @lszomoru ! Any reason why the older request that got enough votes (this one, mine) closed as duplicate of newer feature request that didn't get enough votes yet?

Why it was implement due to the new one and not this one?

I'm asking to understand if something is wrong with my request

**Comment by lszomoru** [comment]
@rhalaly, there was nothing wrong with the feature request that you have filed. Even though I labeled the issue when it was filed in January, I simply forgot about it when https://github.com/microsoft/vscode/issues/148783 was created so I have used that to track the implementation. I only remembered about this issue couple of days ago when a new comment was posted to it. Hope that answers your concerns.
